lucille-plugins/lucille-video/src/main/java/com/kmwllc/lucille/video/VideoFileHandler.java

```diff
@@ -0,0 +1,216 @@
+
+    private final FFmpegFrameGrabber grabber;
+    private final Java2DFrameConverter converter;
+    private final String pathStr;
+    private final String parentId;
+    private final String fileName;
+    private final int frameStride;
+
+    private boolean closed = false;
+    private boolean started = false;
+    private Document nextDoc = null;
+    private int videoFrameIndex = 0;
+
+    FrameDocumentIterator(FFmpegFrameGrabber grabber,
+        Java2DFrameConverter converter,
+        String pathStr,
+        String parentId,
+        String fileName,
+        int frameStride) {
+      this.grabber = grabber;
+      this.converter = converter;
+      this.pathStr = pathStr;
+      this.parentId = parentId;
+      this.fileName = fileName;
+      this.frameStride = frameStride;
+    }
+
+    private void closeAll() {
+      if (closed) {
+        return;
+      }
+      closed = true;
+
+      try {
+        grabber.stop();
+      } catch (Exception ignored) {}
+
+      try {
+        grabber.release();
+      } catch (Exception ignored) {}
+    }
+
+    private boolean preFetch() {
+      if (closed || nextDoc != null) {
+        return nextDoc != null;
+      }
+
+      if (!started) {
+        try {
+          grabber.start();
+          started = true;
+        } catch (Exception e) {
+          closeAll();
+          throw new RuntimeException("Unable to start FFmpegFrameGrabber for: " + pathStr, e);
+        }
+      }
+
+      try {
+        while (true) {
+          Frame frame = grabber.grabImage();
+
+          if (frame == null) {
+            closeAll();
+            return false;
+          }
+
+          int index = videoFrameIndex++;
+          if (frameStride > 1 && (index % frameStride) != 0) {
+            continue;
+          }
+
+          BufferedImage image = converter.convert(frame);
+          if (image == null) {
+            continue;
+          }
+
+          long timeUs = grabber.getTimestamp();
+          long timeMs = timeUs / 1000L;
+
+          byte[] pngBytes = null;
+          try (ByteArrayOutputStream stream = new ByteArrayOutputStream()) {
+            ImageIO.write(image, "png", stream);
+            stream.flush();
+            pngBytes = stream.toByteArray();
+          } catch (IOException e) {
+            log.warn("PNG encoding failed for {} frame {}: {}", pathStr, index, e.toString());
+          }
+
+          String frameId = parentId + "-f" + index;
+          Document doc = Document.create(frameId);
+          doc.setField("source", pathStr);
+          doc.setField("file_name", fileName);
+          doc.setField("frame_index", index);
+          doc.setField("frame_time_ms", timeMs);
+          doc.setField("frame_timecode", formatTimecode(timeMs));
+          doc.setField("image_width", image.getWidth());
+          doc.setField("image_height", image.getHeight());
+          doc.setField("frame_image", pngBytes);
```

Comment:
I think we should either make these field names configurable and/or document what they are being set to. In other FileHandlers (ex CSVFileHandler) all of the fields are configurable. I would definitely lean towards them being configurable with documented defaults, but since there are so many It be ok to just document them as is.